net6_bread_be_tests/Controllers/country_controller_tests.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Xunit;
+using net6_bread_be.Controllers;
+using net6_bread_be.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using net6_bread_be;
+
+public class CountryControllerTests
+{
+    [Fact]
+    public async Task GetAllCountries_ReturnAllCountries_WithInMemoryDatabse()
+    {
+        //Arrange
+        //var dbName = $"TestDatabase_{Guid.NewGuid()}";
+        var options = new DbContextOptionsBuilder<CountryTrackerContext>().UseInMemoryDatabase(databaseName: "CountryDatabase").Options;
+
+        //Add Seeded data into the InMemory db
+        using (var context = new CountryTrackerContext(options))
+        {
+            context.Countries.Add(new Country { CountryId = 1, Name = "Scotland", Description = "Land of the Scots" });
+            context.Countries.Add(new Country { CountryId = 2, Name = "America", Description = "Land of the Free" });
+            await context.SaveChangesAsync();
+        }
+
+        using (var context = new CountryTrackerContext(options))
+        {
+            var controller = new CountryController(context);
+
+            //Act
+            var result = await controller.GetAllCountries();
+
+            //Assert - Making sure that the result isn't null
```

Comment:
This is great, thanks for adding the commented out notes as we continue to get more comfortable with this framework